src/components/UncontrolledTable.tsx

```diff
@@ -0,0 +1,269 @@
+  if (prevSelected !== selected) {
+    setSelectedInternal(forceArray(selected));
+    setPrevSelected(selected);
+  }
+
+  const [pageInternal, setPageInternal] = useState(page);
+  // If the rows prop changes, we need to revert to page 0.
+  const [prevRows, setPrevRows] = useState(rows);
+  if (prevRows !== rows) {
+    setPageInternal(0);
+    setPrevRows(rows);
+  }
+
+  // If the selectable prop changes, the selections stored in state need to be cleared
+  const [prevSelectable, setPrevSelectable] = useState(selectable);
+  if (prevSelectable !== selectable) {
+    setSelectedInternal([]);
+    setPrevSelectable(selectable);
+  }
+
+  // If the expandable prop changes, the expanded rows stored in state need to be cleared
+  const [prevExpandable, setPrevExpandable] = useState(expandable);
+  if (prevExpandable !== expandable) {
+    setExpandedInternal([]);
+    setPrevExpandable(expandable);
+  }
+
+  // Add sortable properties to the columns that are configured as sortable
+  const getSortableColumns = (): SortableColumn<T>[] =>
+    columns
+      .filter((col: SortableColumn<T>) => !col.hidden)
+      .map((col: SortableColumn<T>) => {
+        if (col.sortable !== false) {
+          return {
+            active: sortInternal.column === col.key,
+            ascending: sortInternal.ascending,
+            onSort: (ascending: boolean) => {
+              const column = col.key;
+              let newSort;
+              if (sortInternal.column === column) {
+                newSort = { column, ascending };
+              } else {
+                newSort = { column, ascending: true };
+              }
+              setSortInternal(newSort);
+              onSort(newSort);
+            },
+            sortable: true,
+            ...col,
+          };
+        }
+        return col;
+      });
+
+  // Get the rows that are currently visible based on the combination of pagination and sorting
+  const getVisibleRows = useCallback((): SortableTableRow<T>[] => {
+    const { ascending, column } = sortInternal;
+    const sortedRows = orderBy(
+      rows,
+      [column],
+      [ascending ? 'asc' : 'desc']
+    ) as SortableTableRow<T>[];
+    if (paginated) {
+      const start = pageInternal * pageSize;
+      const end = start + pageSize;
+      return sortedRows.slice(start, end);
+    }
+    return sortedRows;
+  }, [pageInternal, pageSize, paginated, rows, sortInternal]);
+
+  // Any of the prop changes that affect the rows that are currently visible
+  //  will cause the onVisibleRowsChange callback to be called
+  useEffect(() => {
+    onVisibleRowsChange(getVisibleRows());
+  }, [
+    pageInternal,
+    sort,
+    paginated,
+    pageSize,
+    rows,
+    getVisibleRows,
+    onVisibleRowsChange,
+  ]);
+
+  const handlePageChange = useCallback(
+    (current) => {
+      const newPage = current - 1;
+      setPageInternal(newPage);
+      onPageChange(newPage);
+    },
+    [onPageChange, setPageInternal]
```

Comment:
I think `setPageInternal` can be removed here, the linter rule should be smart enough to not warn about this since it's from a `useState` call